File: Youtube_python/channel.py
from datetime import date
from id_generator import IdGenerator
from video import Video

class Channel:
    def __init__(self):
        self.channel_id = IdGenerator.generate_channel_id()
        self.name_channel = ""
        self.host_name = ""
        self.followers = 0
        self.videos_number = 0
        self.date_creation = date.today()
        self.nbre_video = []
    
    @staticmethod
    def create_channel(name, host, creation_date):
        channel = Channel()
        channel.name_channel = name
        channel.host_name = host
        channel.date_creation = creation_date
        return channel
    
    def display_list_video(self):
        if not self.nbre_video:
            print("Aucune vidéo dans cette chaîne.")
            return
        
        for v in self.nbre_video:
            print("\n----------------------------------------")
            print(f"ID Vidéo: {v.video_id}")
            print(f"Titre de la Video: {v.titre_video}")
            print(f"Author: {v.author}")
            print(f"Description: {v.description}")
            print(f"Resolution: {v.resolution}K")
            print(f"Date: {v.date_publication}")
            print(f"Nbre comment: {v.nbre_comment}")
            print(f"Nbre vue: {v.nbre_view}")
            print(f"Likes: {v.nbre_likes}")
            print("----------------------------------------")

File: Youtube_c++/Channel.cpp
#include "Channel.h"
#include "IdGenerator.h"
#include <iostream>
#include <ctime>
#include <sstream>
#include <iomanip>

Channel::Channel() : followers(0), videosNumber(0) {
    channelId = IdGenerator::generateChannelId();
    time_t now = time(0);
    tm* ltm = localtime(&now);
    std::ostringstream oss;
    oss << (1900 + ltm->tm_year) << "-" 
        << std::setw(2) << std::setfill('0') << (1 + ltm->tm_mon) << "-" 
        << std::setw(2) << std::setfill('0') << ltm->tm_mday;
    dateCreation = oss.str();
}

Channel Channel::createChannel(const std::string& name, const std::string& host, 
                              const std::string& date) {
    Channel channel;
    channel.nameChannel = name;
    channel.hostName = host;
    channel.dateCreation = date;
    return channel;
}

void Channel::displayListVideo() const {
    if (nbreVideo.empty()) {
        std::cout << "Aucune vidéo dans cette chaîne.\n";
        return;
    }
    
    for (const auto& v : nbreVideo) {
        std::cout << "\n----------------------------------------\n";
        std::cout << "ID Vidéo: " << v.videoId << "\n";
        std::cout << "Titre de la Video: " << v.titreVideo << "\n";
        std::cout << "Author: " << v.author << "\n";
        std::cout << "Description: " << v.description << "\n";
        std::cout << "Resolution: " << v.resolution << "K\n";
        std::cout << "Date: " << v.datePublication << "\n";
        std::cout << "Nbre comment: " << v.nbreComment << "\n";
        std::cout << "Nbre vue: " << v.nbreView << "\n";
        std::cout << "Likes: " << v.nbreLikes << "\n";
        std::cout << "----------------------------------------\n";
    }
}
